include/cudaq/Optimizer/Dialect/Quake/QuakeInterfaces.td

```diff
@@ -81,4 +81,35 @@ def OperatorInterface : OpInterface<"OperatorInterface"> {
   ];
 }
 
+def MeasurementInterface : OpInterface<"MeasurementInterface"> {
+  let description = [{
+    A measurement operation takes a qubit (qudit) as input, measures it, and
+    returns a classical value, resp. a bit (bitstring). A measurement "destroys"
+    the value of the qubit in the sense the information that the qubit encoded
+    in its state is not recoverable.
+  }];
+  let cppNamespace = "quake";
+
+  let methods = [
+    InterfaceMethod<
+      /*desc=*/ "Returns the set of qubits to be measured",
+      /*retType=*/ "mlir::Operation::operand_range",
+      /*methodName=*/ "getTargets",
+      /*args=*/ (ins),
+      /*methodBody=*/ "return $_op.getTargets();"
+    >,
+    InterfaceMethod<
+      /*desc=*/ "Returns the register name, if any.",
+      /*retType=*/ "std::optional<std::string>",
+      /*methodName=*/ "getOptRegisterName",
```

Comment:
I might suggest `getOptionalRegisterName`, I was thrown off a bit by `Opt` thought it was a typo.